packages/app/ui/components/FacePile/FacePile.tsx

```diff
@@ -0,0 +1,53 @@
+import * as db from '@tloncorp/shared/db';
+import { Text } from '@tloncorp/ui';
+import React from 'react';
+import { View, XStack } from 'tamagui';
+
+import { ContactAvatar } from '../Avatar';
+
+export interface FacePileProps {
+  contacts: db.Contact[];
+  maxVisible?: number;
+}
+
+export const FacePile = React.memo(function FacePileComponent({
+  contacts,
+  maxVisible = 4,
+}: FacePileProps) {
+  const visibleContacts = contacts.slice(0, maxVisible);
```

Comment:
`.slice()` will allocate a new array on each render, which is small but worth a `useMemo` imo (especially if we'll use this on a big list of chats or something)

(this _might_ cause re-renders of user avatar SVGs as well? which would be very expensive in bulk. I'm like 70% sure that React is smarter than that.)

```suggestion
  const visibleContacts = useMemo(() => contacts.slice(0, maxVisible), [contacts, maxVisible]);
```

(you might need to fix prettier/lint if accepting that)